SampleOfferWithoutReward.sol

```diff
@@ -131,6 +131,10 @@ contract SampleOfferWithoutReward {
     // Executing this function before the Offer is signed off by the Client
     // makes no sense as this contract has no money.
     function getDailyPayment() noEther {
+        if (msg.sender != contractor && msg.sender != address(client)) {
```

Comment:
I guess the `msg.sender` is not changes because solidity implements internal calls without the `CALL` instruction.